files/build-files-from-ncbi.py

```diff
@@ -52,6 +53,38 @@ def get_genome_row(genome_info):
 def get_genomes_df(tax_ids):
   return pd.DataFrame(data=[get_genome_row(genome_info) for genome_info in requests.get(build_genomes_url(tax_ids)).json()["reports"]])
 
+def _id_to_gene_model_url(asm_id):
+  hubs_url = "https://hgdownload.soe.ucsc.edu/hubs/"
+  components = [asm_id[0:3], asm_id[4:7], asm_id[7:10], asm_id[10:13], asm_id, "genes"]
+  url = urllib.parse.urljoin(hubs_url, "/".join(components))
+  # url looks something like https://hgdownload.soe.ucsc.edu/hubs/GCF/030/504/385/GCF_030504385.1/genes/
+  # and contains html content with links to gene models.
+  # we need to scrape this to get the gtf
+  print(f"fetching url {url}")
+  response = requests.get(url)
+  try:
+    response.raise_for_status()
+  except Exception:
+    # FIXME?: Some accessions don't have a gene folder
+    return None
+  # find link to gtf, should ideally be ncbiRefSeq, but augustus will do
+  html_content = response.text
+  pattern = rf"{asm_id.replace('.', r'\.')}.*?\.gtf\.gz"
+  augustus_file = None
+  for match in re.findall(pattern, html_content):
+    if "ncbiRefSeq" in match:
+      return urllib.parse.urljoin(f"{url}/", match)
+    elif "augustus" in match:
+      augustus_file = match
+  if augustus_file:
+    return urllib.parse.urljoin(f"{url}/", augustus_file)
+  # No match, I guess that's OK ?
+  return None
+
+
+def add_gene_model_url(genomes_df: pd.DataFrame):
+  return pd.concat([genomes_df, genomes_df["accession"].apply(_id_to_gene_model_url).rename("geneModelUrl")], axis="columns")
```

Comment:
I've added gene model URL -- not certain whether `accession` is appropriate for getting that